### ISSUE TITLE
[RFC] refactor testapi to support multiple groups

### ISSUE THREAD

**Issue by caesarxuchao** [kubernetes/kubernetes#13233]
Our tests depend on the package `testapi` and the `KUBE_API_VERSION` environment variable to get version and codec etc. As we introduce API groups, which may have individual versioning systems, we need to rethink how to set the test versions.

Here is my proposal:

* define a env var `KUBE_TEST_VERSIONS` to express what version of a group is going to be tested, e.g. KUBE_TEST_VERSIONS="v1,exprimental/v1alph1". 

* let the functions in [testapi.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/testapi/testapi.go) take `group string`as input, and loop through KUBE_TEST_VERSIONS to look for the correct version. (Implementation-wise, we can define the input to be `group ...string` so that we don't need to modify the existing calls to these functions.)

One caveat:
1. if we want to run tests of one group for multiple versions (e.g. experimental/v1alpha1 and experimental/v1alpha2), we have to rerun all other groups as well, i.e.,
```
KUBE_TEST_VERSIONS="v1,experimental/v1alpha1" go test
KUBE_TEST_VERSIONS="v1,experimental/v1alpha2" go test
```

Did I miss something? Thanks

@nikhiljindal @lavalamp

**Comment by lavalamp** [comment]
Yeah, that sounds about right. Perhaps KUBE_TEST_API_GROUP_DEFAULTS or just KUBE_TEST_API_DEFAULTS for the env variable name?

**Comment by caesarxuchao** [comment]
I think DEFAULTS is misleading. The env variable is the only way to set which version to be test, so it's not DEFAULTS. Tests should not set versions, they should call testapi functions to figure out which version is being tested.

**Comment by lavalamp** [comment]
@caesarxuchao Can we close this now that your testing PR got merged?

**Comment by caesarxuchao** [comment]
Yes, it's fixed by #13581. Thanks for the reminding.
